sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -528,3 +530,20 @@ case class SetDatabaseCommand(databaseName: String) extends RunnableCommand {
 
   override val output: Seq[Attribute] = Seq.empty
 }
+
+
+/**
+ * Returns the current database of metadataHive.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_() - Returns the current database.",
+  extended = "> SELECT _FUNC_()")
+private[sql] case class CurrentDatabase(ctx: SQLContext)
```

Comment:
i'm afraid this is too convoluted -- why don't we just have this function doesn't take any arguments, and then in the optimizer (which should take a SessionState) replace it with a literal? similar to how we compute CurrentDate.